src/main/java/org/folio/innreach/external/service/impl/InnReachContributionServiceImpl.java

```diff
@@ -46,32 +47,54 @@ public InnReachResponse contributeBib(UUID centralServerId, String bibId, BibInf
+      var connectionDetails = getConnectionDetails(centralServerId);
+
+      var accessTokenDTO = innReachAuthExternalService.getAccessToken(connectionDetails);
+      var connectionUrl = URI.create(connectionDetails.getConnectionUrl());
+      var authorizationHeader = buildBearerAuthHeader(accessTokenDTO.getAccessToken());
+      var localCode = connectionDetails.getLocalCode();
+      var centralCode = connectionDetails.getCentralCode();
+
+      var response = contributionClient.deContributeBib(connectionUrl, authorizationHeader, localCode,
+        centralCode, bibId);
+      if (!response.getErrors().isEmpty()) {
+        var error = response.getErrors().get(0).getReason();
+        if (error.contains("Contribution to d2irm is currently suspended")) {
+          throw new ServiceSuspendedException("Contribution to d2irm is currently suspended");
+        }
+      }
+      return  response;
+    } catch (ServiceSuspendedException ex) {
+      throw new ServiceSuspendedException(ex.getMessage());
+    }
   }
 
   @Override
   public InnReachResponse deContributeBibItem(UUID centralServerId, String itemId) {
-    log.debug("deContributeBibItem:: parameters centralServerId: {}, itemId: {}", centralServerId, itemId);
-    var connectionDetails = getConnectionDetails(centralServerId);
-
-    var accessTokenDTO = innReachAuthExternalService.getAccessToken(connectionDetails);
-    var connectionUrl = URI.create(connectionDetails.getConnectionUrl());
-    var authorizationHeader = buildBearerAuthHeader(accessTokenDTO.getAccessToken());
-    var localCode = connectionDetails.getLocalCode();
-    var centralCode = connectionDetails.getCentralCode();
-
-    return contributionClient.deContributeBibItem(connectionUrl, authorizationHeader, localCode,
-      centralCode, itemId);
+    try {
+      log.debug("deContributeBibItem:: parameters centralServerId: {}, itemId: {}", centralServerId, itemId);
+      var connectionDetails = getConnectionDetails(centralServerId);
+
+      var accessTokenDTO = innReachAuthExternalService.getAccessToken(connectionDetails);
+      var connectionUrl = URI.create(connectionDetails.getConnectionUrl());
+      var authorizationHeader = buildBearerAuthHeader(accessTokenDTO.getAccessToken());
+      var localCode = connectionDetails.getLocalCode();
+      var centralCode = connectionDetails.getCentralCode();
+
+      var response = contributionClient.deContributeBibItem(connectionUrl, authorizationHeader, localCode,
+        centralCode, itemId);
+      if (!response.getErrors().isEmpty()) {
+        var error = response.getErrors().get(0).getReason();
+        if (error.contains("Contribution to d2irm is currently suspended")) {
+          throw new ServiceSuspendedException("Contribution to d2irm is currently suspended");
+        }
+      }
+      return response;
+    } catch (ServiceSuspendedException ex) {
+      throw new ServiceSuspendedException(ex.getMessage());
+    }
```

Comment:
This is nearly the exact same code block as in `deContributeBib` above. Consolidate into a shared method.